### PR TITLE
CCO-358: Manifest changes necessary to support Azure Workload Identity

### PR DESCRIPTION
**Pull request by dlom**
xref: [CCO-358](https://issues.redhat.com//browse/CCO-358)
xref: [CCO-294](https://issues.redhat.com//browse/CCO-294)

* Enhancement proposal: https://github.com/openshift/enhancements/pull/1301
* Corresponding change for `cloud-network-config-controller`: https://github.com/openshift/cloud-network-config-controller/pull/102

### CODE REVIEW

File: manifests/0000_70_cluster-network-operator_01_credentialsrequest.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ metadata:
     include.release.openshift.io/self-managed-high-availability: "true"
     include.release.openshift.io/single-node-developer: "true"
 spec:
+  serviceAccountNames:
```
Comment (kyrtapz) [inline manifests/0000_70_cluster-network-operator_01_credentialsrequest.yaml:12]: Is this the right file?
`cloud-network-config-controller` credential requests are stored in `manifests/02-cncc-credentials.yaml` and already have the `serviceAccountNames` set.

Comment (dlom) [reply manifests/0000_70_cluster-network-operator_01_credentialsrequest.yaml]: Good catch, thanks

**Comment by dlom** [comment]
@kyrtapz is there a better way to ensure that the tests pass besides spamming retry and hoping for the best? There's no actual code changes in this PR so presumably all the tests should pass if they passed on a previous PR

**Comment by kyrtapz** [comment]
> @kyrtapz is there a better way to ensure that the tests pass besides spamming retry and hoping for the best? There's no actual code changes in this PR so presumably all the tests should pass if they passed on a previous PR

I don't think there is a better way unfortunately. We should review whether the failures are flakes before spamming retries though :)
We need to make sure that the required tests are passing, there was a bug that caused the migration tests to perma-fail which should be fixed already.

**Comment by jcaamano** [comment]
The e2e-aws-ovn-network-migration and e2e-vsphere-ovn-windows jobs are permafailing, @kyrtapz opened two bugs for them respectively:
https://issues.redhat.com/browse/OCPBUGS-14470
https://issues.redhat.com/browse/OCPBUGS-14469

This change should be unrelated with the failures.

/override ci/prow/e2e-aws-ovn-network-migration
/override ci/prow/e2e-vsphere-ovn-windows

**Comment by dlom** [comment]
@kyrtapz @jcaamano looks like the last remaining merge blocker is a valid jira bug.  I already have a valid jira reference; what's the difference between the two?

**Comment by kyrtapz** [comment]
Feature merging is currently blocked in master and you need to work with TRT to get a feature merged:
https://docs.google.com/document/d/1_0qUViEqaSSpMOPElK_tGrzSCGphMnkqfQw6vIbzbEM

**Comment by dlom** [comment]
Submitted PR to the feature queue

**Pull request merged**